### PR TITLE
User agreement box

### PR DESCRIPTION
**Pull request by bhatiadheeraj**

### CODE REVIEW

File: ui/src/projects.vue [hunk 1]
```diff
@@ -221,6 +226,37 @@ export default {
...
+                // User has not accepted the latest version of the data use agreement
+                if (userAgreementDate < latestAgreementDate) {
+                    console.log("User has not accepted the latest version of the data use agreement", userAgreementDate, latestAgreementDate);
+                    return false;
+                }
+
+                return true;
+            }
+           
+            return true;
```
Comment (anibalsolon) [inline ui/src/projects.vue:258]: shouldnt this be return false?

Comment (bhatiadheeraj) [reply ui/src/projects.vue:258]: brainlife projects page can be accessed without logging in so it handles that case

Comment (anibalsolon) [reply ui/src/projects.vue:258]: ok I'd suggest two things:
- move the `userAgreementAgreed` logic to inside the modal component, to keep it in one place, and show the modal on mounted if the logic is true
- ` v-if=` the modal to check if the user is logged. calling it `userAgreementAgreed` and returning true for non-logged users is misleading

File: ui/src/projects.vue [hunk 2]
```diff
@@ -44,7 +44,8 @@
 
             <!--TODO - should refactor this.. similar to public y projects-->
             <div v-if="config.user" class="position: relative">
-                <h4 class="group-title">My Projects</h4>
+                <aupAgreementModal v-if="config.user && config.profile"></aupAgreementModal>
```
Comment (bhatiadheeraj) [inline ui/src/projects.vue:47]: how about only checking for `config.profile ?`

Comment (anibalsolon) [reply ui/src/projects.vue:47]: I'd keep user as well, just to make sure heh

**Review by anibalsolon — APPROVED** [review]

File: ui/src/projects.vue (hunk 1 above)
Comment (anibalsolon) [reply ui/src/projects.vue]: yeah! thank you

**Pull request merged**